src/components/TodoApp.tsx

```diff
@@ -0,0 +1,157 @@
+import { useEffect, useState } from 'react';
+
+import { TodoHeader } from './TodoHeader';
+import { TodoList } from './TodoList';
+import { TodoFooter } from './TodoFooter';
+import { TodoError } from './TodoError';
+
+// import { createTodo, getTodos, USER_ID, deleteTodo } from '../api/todos';
+import * as todoService from '../api/todos';
+import { Todo } from '../types/Todo';
+import { Filters } from '../types/Filter';
+import { MessageError } from '../types/ErrorMessage';
+
+export const TodoApp = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.All);
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<MessageError>(
+    MessageError.default,
+  );
+  const [loading, setLoading] = useState(false);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setIsError(true);
+        setErrorMessage(MessageError.loadError);
+      });
+  }, []);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (isError) {
+        setIsError(false);
+        setErrorMessage(MessageError.default);
+      }
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [isError]);
+
+  function addTodo(str: string) {
+    setLoading(true);
+
+    setTempTodo({
+      id: 0,
+      userId: todoService.USER_ID,
+      title: str.trim(),
+      completed: false,
+    });
+
+    return todoService
+      .createTodo(str)
+      .then(todo => {
+        setTodos(currentTodos => [...currentTodos, todo]);
+
+        return true;
+      })
+      .catch(() => {
+        setIsError(true);
+        setErrorMessage(MessageError.addError);
+
+        return false;
+      })
+      .finally(() => {
+        setLoading(false);
+        setTempTodo(null);
+      });
+  }
+
+  function deleteTodo(todoId: number) {
```

Comment:
It is important to follow the same way of creating functions, fix it everywhere
```suggestion
  const deleteTodo = (todoId: number) => {
```